app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -168,3 +199,16 @@ export const addListenSignalData = (payload) => {
     ],
   });
 };
+
+export const updateListenSignalData = (payload, signalId) => {
+  return createRequestAction({
+    endpoint: listenSignalEndpoint(signalId),
+    method: 'PUT',
+    body: JSON.stringify(payload),
+    types: [
+      LISTEN_SIGNALS_PUT_REQUEST,
+      LISTEN_SIGNALS_PUT_REQUEST_SUCCESS,
+      LISTEN_SIGNALS_PUT_REQUEST_FAIL,
```

Comment:
You should pass in the signal id as a meta prop.
```js
{ type: LISTEN_SIGNALS_PUT_REQUEST_FAIL, meta: { id: signalId } },
```

You're calling it within the reducer function so it'll fail.